### PR TITLE
pkp/pkp-lib#4398: Ensure user is persisted before trying assignUserToGroup()

### PR DESCRIPTION
**Pull request by ctgraham**
Resolves unset variable warning for pkp/pkp-lib#4398

### CODE REVIEW

**Comment by bozana** [comment]
@ctgraham, yes, this is good, but I think we should also:
-- add an error, because it is a mismatch -- both username and e-mail are unique so it should be one and the same user
-- it seems to me that the USER_REGISTER e-mail should only be sent if a new user is inserted, or? -- If it is an existing user (with that username and e-mail) only the groups would be added to him/her, all other metadata are not changed, or? If so, maybe the e-mail should not be sent? What do you think?

If you agree I can add those...

**Comment by bozana** [comment]
@asmecher, it seems to me that the USER_REGISTER e-mail should only be sent if a new user is being imported, or? -- Looking into the code, if the import XML contains an existing user (with that username and e-mail) only the groups would be added to him/her, all other metadata are not changed, or? If so, maybe the e-mail should not be sent in that case? What do you think?

**Comment by asmecher** [comment]
It's common practice to send a "welcome" message after a user account is created (also to notify someone e.g. if their email has been misused by someone else). But per https://github.com/pkp/pkp-lib/issues/2135 we should definitely not be emailing passwords. In case it's a manager-created user we mitigate the risk by making it a one-time-use password (requiring reset upon the first login), but if it's a user who has registered their own account, a password is definitely inappropriate.

TL;DR: `USER_REGISTER` can be used for both self-registrations and manager-created accounts, but for self-registrations it should definitely not be sending passwords. So I see two approaches:
- Do not send a welcome email for self-registrations, or
- Create another email template for self-registrations that does not include a password.

I'd favour the second approach, on balance.

**Comment by bozana** [comment]
Well... I actually thought something else:
When importing user, we look if that user already exists, s. https://github.com/pkp/pkp-lib/blob/stable-3_1_2/plugins/importexport/users/filter/UserXmlPKPUserFilter.inc.php#L157-L159. In that case we also send the e-mail, as far as I can see: https://github.com/pkp/pkp-lib/blob/stable-3_1_2/plugins/importexport/users/filter/UserXmlPKPUserFilter.inc.php#L248-L255. So I was wondering why we would like to send that e-mail in that case i.e. if this is intended/correct behavior :-)
But, now you mentioned another thing/issue, that I will then take a look separately... :-)
